File: core/src/commonMain/kotlin/io/umain/munchies/core/TextIdHelper.kt
package io.umain.munchies.core

/**
 * Maps TextId sealed class instances to their corresponding translation keys.
 * This provides a single source of truth for text resource mapping.
 */
fun mapTextIdToKey(textId: TextId): String = when (textId) {
    TextId.AppTitle -> "app.title"
    
    TextId.RestaurantListTitle -> "restaurant.list.title"
    TextId.RestaurantDetailTitle -> "restaurant.detail.title"
    TextId.FilterAll -> "filter.all"
    
    TextId.RestaurantStatusOpen -> "restaurant.status.open"
    TextId.RestaurantStatusClosed -> "restaurant.status.closed"
    
    TextId.AccessibilityRestaurantCard -> "accessibility.restaurant.card"
    TextId.AccessibilityFilterChip -> "accessibility.filter.chip"
    TextId.AccessibilityFilterSelected -> "accessibility.filter.selected"
    TextId.AccessibilityBackButton -> "accessibility.back.button"
    
    TextId.ErrorLoading -> "error.loading"
    TextId.ErrorNetwork -> "error.network"
    TextId.Loading -> "loading"
}

File: iosApp/iosApp/Core/Localization/TextIdHelper.swift
import Foundation
import shared

func tr(_ textId: TextId, _ args: Any...) -> String {
    return TranslationKt.tr(textId: textId, args: KotlinArray<AnyObject>(size: Int32(args.count)) { index in
        args[Int(truncating: index)] as AnyObject
    })
}

extension TextId {
    var localized: String {
        return tr(self)
    }
}

extension TextId {
    static var appTitle: TextId { TextId.AppTitle() }
    static var restaurantListTitle: TextId { TextId.RestaurantListTitle() }
    static var restaurantDetailTitle: TextId { TextId.RestaurantDetailTitle() }
    static var filterAll: TextId { TextId.FilterAll() }
    static var restaurantStatusOpen: TextId { TextId.RestaurantStatusOpen() }
    static var restaurantStatusClosed: TextId { TextId.RestaurantStatusClosed() }
    static var accessibilityRestaurantCard: TextId { TextId.AccessibilityRestaurantCard() }
    static var accessibilityFilterChip: TextId { TextId.AccessibilityFilterChip() }
    static var accessibilityFilterSelected: TextId { TextId.AccessibilityFilterSelected() }
    static var accessibilityBackButton: TextId { TextId.AccessibilityBackButton() }
    static var errorLoading: TextId { TextId.ErrorLoading() }
    static var errorNetwork: TextId { TextId.ErrorNetwork() }
    static var loading: TextId { TextId.Loading() }
}
